sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1910,486 @@ def transform(self, X):
+        If None, the random number generator is the RandomState instance used
+        by np.random.
+
+    Attributes
+    ----------
+    quantiles_ : ndarray, shape (n_quantiles_, n_features)
+        The values corresponding the quantiles of reference.
+
+    Examples
+    --------
+    >>> from sklearn.datasets import load_iris
+    >>> from sklearn.preprocessing import QuantileTransformer
+    >>> iris = load_iris()
+    >>> X_trans = QuantileTransformer(n_quantiles=20).fit_transform(iris.data)
+
+    See also
+    --------
+    quantile_transform : Equivalent function without the object oriented API.
+
+    StandardScaler : perform standardization that is faster, but less robust
+    to outliers.
+
+    RobustScaler : perform robust standardization that removes the influence
+    of outliers but does not put outliers and inliers on the same scale.
+    """
+
+    def __init__(self, n_quantiles=1000, output_distribution='uniform',
+                 ignore_implicit_zeros=False, subsample=int(1e5),
+                 smoothing_noise=None, random_state=None):
+        self.n_quantiles = n_quantiles
+        self.output_distribution = output_distribution
+        self.ignore_implicit_zeros = ignore_implicit_zeros
+        self.subsample = subsample
+        self.smoothing_noise = smoothing_noise
+        self.random_state = random_state
+
+    def _compute_quantile_one_column(self, X_col, references, random_state):
+        """Private function to compute the quantiles for one features."""
+        if self.smoothing_noise is None:
+            noise = np.zeros(X_col.shape)
+        else:
+            noise = random_state.normal(0, self.smoothing_noise,
+                                        size=X_col.shape)
+
+        return np.percentile(X_col + noise, references)
+
+    def _dense_fit(self, X, random_state):
+        """Compute percentiles for dense matrices.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            The data used to scale along the features axis.
+        """
+        if self.ignore_implicit_zeros:
+            warnings.warn("'ignore_implicit_zeros' takes effect only with"
+                          " sparse matrix. This parameter has no effect.")
+
+        # subsample the matrix X if necessary
+        n_samples, n_features = X.shape
+        if self.subsample < n_samples:
+            subsample_idx = choice(range(n_samples), size=self.subsample,
+                                   replace=False, random_state=random_state)
+        else:
+            subsample_idx = slice(None)
+
+        # for compatibility issue with numpy<=1.8.X, references
+        # need to be a list scaled between 0 and 100
+        references = np.linspace(0, 100, self.n_quantiles,
+                                 endpoint=True).tolist()
+        self.quantiles_ = np.transpose(
+            [self._compute_quantile_one_column(X[subsample_idx, feature_idx],
```

Comment:
Do you mean from the perspective of the code, or from the perspective of the model?